### DISCUSSION TITLE
Processing credit card details in an Edge or API Function?

### DISCUSSION THREAD

**Discussion by caleb-kwp** [vercel/community#4496] [Feature Question]
### Question

Hi,

My Payment Provider's API SDK has a method to convert a credit card into a payment token.
I don't want to expose the API Credentials to the front-end. 
Meaning I either need to host my own PCI Compliant server to process credit cards, or use a Vercel Edge function.

I would rather not have to mess with PCI Compliance, so I'm wondering if it's possible/permitted/compliant to send credit cards to a Vercel Edge or API Function (in the app/ directory)?

Many thanks

**Comment by leerob** [comment]
We would recommend using an iframe to your payment processor here: https://vercel.com/blog/pci-compliance-for-ecommerce-teams
